pkg/gatewayserver/gatewayserver.go

```diff
@@ -897,16 +899,19 @@ func (gs *GatewayServer) handleUpstream(ctx context.Context, conn connectionEntr
 			if msg.Message.Payload == nil {
 				msg.Message.Payload = &ttnpb.Message{}
 				if err := lorawan.UnmarshalMessage(msg.Message.RawPayload, msg.Message.Payload); err != nil {
-					registerDropUplink(ctx, gtw, msg, "validation", err)
```

Comment:
Because it's before the receive.

Traditionally we had first receive, then drop or forward in GS, NS and AS. In GS, that drop or forward is per upstream. This way we wouldn't miss traffic in our metrics.

When reusing this `gs.up.drop` for validation _before_ a receive, the order is off, `count(upstreams) * receives != drops + forwards`, and we get a weird label value (`validation`) for host. I get that it's nice to have something here, but it's not clean. That is the reason this one gets removed here.

---

I think we need to decide what we want to achieve. I do value some sort of orderly events and also consistency. What we could do is:

1. Receive: everything, raw payload only. Could be non-LoRaWAN, could be CRC failure, could be unsupported LoRaWAN major version, etc
2. Parse and validation errors; new event. Includes CRC failure
3. Accept; new event
4. Forward and drop: per upstream only

Or we keep it simple:

1. Receive: only something that makes sense, i.e. something that passes LoRaWAN parsing. That is only relevant anyway
2. Drop if CRC or validation fails
3. Forward and drop: per upstream

This is what is implemented now. This is not entirely clean, but we wouldn't have to refactor the events and we stay close to what we have in NS and AS.

Reply:
I am ok with keeping option 2 (what is implemented), and doing option 1 for the next minor. 

In general I am not sure if `receive` and fork to `drop` or `forward` is really consistent, at least body wise, between components, but I am ok with revisiting this.